src/components/tl/mlx5/tl_mlx5_dm.c

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5_dm.h"
+
+static ucc_status_t ucc_tl_mlx5_dm_chunk_alloc(ucc_mpool_t *mp, //NOLINT
+                                               size_t *size_p, void **chunk_p)
+{
+    *chunk_p = ucc_malloc(*size_p, "mlx5 dm");
+    if (!*chunk_p) {
+        return UCC_ERR_NO_MEMORY;
+    }
+    return UCC_OK;
+}
+
+static void ucc_tl_mlx5_dm_chunk_init(ucc_mpool_t *mp,        //NOLINT
+                                      void *obj, void *chunk) //NOLINT
+{
+    ucc_tl_mlx5_dm_chunk_t *c    = (ucc_tl_mlx5_dm_chunk_t *)obj;
+    ucc_tl_mlx5_team_t     *team =
+        ucc_container_of(mp, ucc_tl_mlx5_team_t, dm_pool);
+
+    c->offset                = (ptrdiff_t)team->oob_req;
```

Comment:
why call it oob_req? isn't it just offset in dm allocation?

Reply:
changed to dm_offset